composer/src/Plugin.php

```diff
@@ -55,8 +55,12 @@ public function addVagrantfile(Event $event) {
         $source = __DIR__ . '/../../.beetbox/Vagrantfile';
         $target =  $baseDir . '/Vagrantfile';
 
-        if (!file_exists($target) || md5_file($source) != md5_file($target)) {
-            copy($source, $target);
+        if (file_exists($source)) {
+            if (!file_exists($target) || md5_file($source) != md5_file($target)) {
+                copy($source, $target);
+            }
+        } elseif (file_exists($target)) {
```

Comment:
Removing `Vagrantfile` like this is too intrusive: it will remove a file modified by user. Can it be removed only if it has not been modified by the user and If it has been modified - leave it in place. 

A possible solution is to store checksum of original `Vagrantfile` somewhere in vagrant cache, read it from there during uninstall and compare to current `Vagrantfile` being unlinked..